### PR TITLE
Update actions used in CI

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,21 +27,12 @@ jobs:
             os: ubuntu-latest
             arch: x64
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: julia-actions/setup-julia@v1
```
Comment (odow) [inline .github/workflows/ci.yml:31]: I learnt after the MOI PR that this should actually be `@latest` or `@v2`. There are still a few deprecation notices haha

Comment (blegat) [reply .github/workflows/ci.yml:31]: `v4` seems to be used in the template: https://github.com/JuliaCI/PkgTemplates.jl/blob/master/templates/github/workflows/CI.yml what's the issue ?

Comment (odow) [reply .github/workflows/ci.yml:31]: No I meant `julia-actions/setup-julia@v2`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,24 +27,15 @@ jobs:
...
-          restore-keys: |
-            ${{ runner.os }}-test-${{ env.cache-name }}-
-            ${{ runner.os }}-test-
-            ${{ runner.os }}-
+      - uses: julia-actions/cache@v1
       - uses: julia-actions/julia-buildpkg@v1
       - uses: julia-actions/julia-runtest@v1
       - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
```
Comment (odow) [inline .github/workflows/ci.yml:39]:
```suggestion
      - uses: codecov/codecov-action@v3
```

**Pull request merged**